Classes.py

```diff
@@ -0,0 +1,340 @@
+        self.mid_row = [_Element() for i in range(3)]
+        self.bottom_row = [_Element() for i in range(3)]
+        self.box_pos = None  # NOT USED
+
+        self.set_element_box()
+        self.set_element_row()
+
+    # Methods run at initialization.
+    def set_element_box(self):  # Sets the box attribute of the element.
+        for element in self:
+            element._box = self
+
+    def set_element_row(self):  # Sets the row attribute of the element.
+        for element in self.top_row:
+            element._row = self.top_row
+        for element in self.mid_row:
+            element._row = self.mid_row
+        for element in self.bottom_row:
+            element._row = self.bottom_row
+
+            # General methods
+
+    def get_index(self, val):  # Returns the index of an element.
+        count = 0
+        for i in self:
+            if i.get_value() == val:
+                return count
+            count += 1
+
+    # Operator overloading methods
+    def __iter__(self):  # functionality -> for x in box:
+        for element in self.top_row:  # Iterates in the same way as a matrix
+            yield element  # i.e. from left to right.
+        for element in self.mid_row:
+            yield element
+        for element in self.bottom_row:
+            yield element
+
+    def __getitem__(self, index):  # functionality -> getting value of box[3]
+        count = 0  # returns value of specified index.
+        for element in self:  # indexing from 0-9; indexing is in the same way
+            if count == index:  # as a matrix.
+                return element
+            count += 1
+
+    def __setitem__(self, index, val):  # functionality -> assignment box[2]=3
+        count = 0  # assignment at a specific index.
+        for element_place in range(9):
+            if count == element_place:
+                self[element_place].set_value(val)
+            count += 1
+
+    def __contains__(self, val):  # functionality -> val in box
+        for element in self:  # returns True or False.
+            if element.get_value() == val:
+                return True
+        return False
+
+
+class _Element:
+    ''' A class that acts as the element(number) in one of the boxes of sudoku. '''
+
+    def __init__(self, value=None):
+        self._box = None  # The box this element belongs to.   #NOT USED
+        self._row = None  # The row this element belongs to.   #NOT USED
+        self._value = value  # The value the element has.
+
+    # getter methods.
+    def get_box(self):  # To access the box this element belongs to.
+        return self._box
+
+    def get_value(self):  # To access the value this element has.
+        return self._value
+
+    def get_row(self):  # To access the row this element belongs to.
+        return self._row
+
+        # setter methods.
+
+    def set_value(self, value):  # To access the value this element will have.
+        self._value = value
+
+
+"""
```

Comment:
Remove all the testing code from commits, if they are relevant make use of the `unittest` module which will be automatically run by the CI as well